### PR TITLE
Deprecate MAV_CMD_REQUEST_* in favor of MAV_CMD_REQUEST_MESSAGE (#1180)

### PR DESCRIPTION
**Pull request by amilcarlucas**
I feel like this is a step in the right direction deprecating a lot of messages in favor of a much simpler protocol.

let the discussions begin

### CODE REVIEW

File: message_definitions/v1.0/minimal.xml [hunk 1]
```diff
@@ -621,7 +621,13 @@
     </enum>
     <enum name="MAV_CMD">
       <description>Commands to be executed by the MAV. They can be executed on user request, or as part of a mission script. If the action is used in a mission, the parameter mapping to the waypoint/mission message is as follows: Param 1, Param 2, Param 3, Param 4, X: Param 5, Y:Param 6, Z:Param 7. This command list is similar what ARINC 424 is for commercial aircraft: A data format how to interpret waypoint/mission data. See https://mavlink.io/en/guide/xml_schema.html#MAV_CMD for information about the structure of the MAV_CMD entries</description>
+      <entry value="512" name="MAV_CMD_REQUEST_MESSAGE" hasLocation="false" isDestination="false">
```
Comment (hamishwillee) [inline message_definitions/v1.0/minimal.xml:624]: We certainly can't do this :-). Arguably though this should be in common.xml, but that should be a separate PR.

**Comment by hamishwillee** [comment]
1. Common.xml should be modified in mavlink/mavlink.
2. I agree. But there was some concern from @dogmaphobic that the camera APIs are hard for him to update - ie the GCS ends up having to send each message twice because it can't (yet) tell which requestor will work.

**Comment by peterbarker** [comment]
On Tue, 24 Nov 2020, Amilcar Lucas wrote:

> I feel like this is a step in the right direction deprecating a lot of
> messages in favor of a much simpler protocol.
> 
> let the discussions begin

I'd have absolutely no problem with this PR if the message hadn't been 
moved to minimal.xml as part of it.

**Comment by hamishwillee** [comment]
> I'd have absolutely no problem with this PR if the message hadn't been moved to minimal.xml as part of it.

@peterbarker MAV_CMD_REQUEST_MESSAGE went into minimal.xml a long while ago on mavlink/mavlink, and this was not questioned until common.xml imported minimal.xml - resulting in MAV_CMD_REQUEST_MESSAGE  being removed from common.xml.

I'm not saying this right, just an FYI and note that it reflects upstream. I'm leaning towards MAV_CMD_REQUEST_MESSAGE moving back into common.xml. It was added because it supports the version handshake https://mavlink.io/en/guide/mavlink_version.html#version_handshaking (ie generic requestor rather than MAV_CMD_REQUEST_PROTOCOL_VERSION.

**Comment by peterbarker** [comment]
@amilcarlucas `MAV_CMD_REQUEST_MESSAGE` is no longer in minimal.xml upstream.

I think this could be merged if the addition of that to minimal.xml was removed.

**Comment by amilcarlucas** [comment]
Points addressed now

**Review by peterbarker — APPROVED** [review]
